service/src/main/java/com/ge/predix/acs/privilege/management/dao/ResourceHierarchicalRepository.java

```diff
@@ -10,4 +10,6 @@ ResourceEntity getResourceWithInheritedAttributes(final ZoneEntity zone,
                                                       final String resourceIdentifier);
 
     Set<String> getResourceEntityAndDescendantsIds(final ResourceEntity entity); 
+    int getVersion();
+    void setVersion(final int version);
```

Comment:
Its a bit strange to have a public setVersion method on a repository. I cant think of an alternate solution though.